calculator.js

```diff
@@ -0,0 +1,65 @@
+var writes = document.getElementById('screen');
+var currNumber = 0;
+var result = 0;
+
+var buttons = document.querySelector('.allButtons');
+var numberButtons = buttons.querySelectorAll('.number');
+
+for (var i = 0; i <= numberButtons.length - 1; i++) {
+
+    numberButtons[i].onclick = function() {
+
+        var buttonNumber = parseInt(this.innerHTML);
+
+        currNumber = currNumber * 10 + buttonNumber
+
+        writes.value = currNumber;
+
+    }
+};
+
+var operationButtons = buttons.querySelectorAll('.operation');
+
+for (var i = 0; i <= operationButtons.length - 1; i++) {
+    var isInitialAction = true;
+    var prevOperation;
+    operationButtons[i].onclick = function() {
+
+        var operation = this.innerHTML;
+
+ /*===================================================================*/
+        if (operation === this.innerHTML) {
+
+            if (isInitialAction) {
+                result = currNumber;
+                currNumber = 0;
+                writes.value = result;
+                isInitialAction = false;
+            }
+        else {
+
+                if (prevOperation === '+') {
```

Comment:
In every IF you have two lines which are repeating:
```
currNumber = 0;
writes.value = result;
```
Since they are going to be executed in any case, why do not you extract them outside IFs:
```
if (prev .. ) { ... }
if (prev .. ) { ... }
...
currNumber = 0;
writes.value = result;
```